pkg/operation/botanist/operatingsystemconfig.go

```diff
@@ -395,6 +395,29 @@ func (b *Botanist) generateCloudConfigExecutionChart() (*chartrenderer.RenderedC
 			w["command"] = *cmd
 		}
 
+		if worker.KubeletDataVolumeName != "" && worker.DataVolumes != nil {
+			for _, dataVolume := range worker.DataVolumes {
+				volName := dataVolume.Name
+				if *volName == worker.KubeletDataVolumeName {
+					size, err := resource.ParseQuantity(dataVolume.Size)
+					if err != nil {
+						return nil, err
+					}
+					sizeInBytes, ok := size.AsInt64()
+					if !ok {
+						sizeInBytes, ok = size.AsDec().Unscaled()
+						if !ok {
+							return nil, fmt.Errorf("failed to parse volume size %s", dataVolume.Size)
+						}
+					}
+					w["kubeletDataVolume"] = map[string]interface{}{
```

Comment:
Can you please add these new values as commented section to the respective Helm chart `values.yaml`?